dotCMS/src/integration-test/java/com/dotcms/security/apps/AppsAPIImplTest.java

```diff
@@ -329,6 +339,84 @@ public void Test_Non_Admin_User_With_Portlet_Read_Attempt() throws DotDataExcept
 
+    @DataProvider
+    public static Object[] getTestCases() throws Exception {
+        return new Object[]{
+                new UTFCharsRangeTestCase(0, 127, "C0 Controls and Basic Latin"),
+                new UTFCharsRangeTestCase(128, 255, "C1 Controls and Latin-1 Supplement"),
+                new UTFCharsRangeTestCase(256, 383, "Latin Extended-A"),
+                new UTFCharsRangeTestCase(384, 591, "Latin Extended-B"),
+                new UTFCharsRangeTestCase(688, 767, "Spacing Modifiers"),
+                new UTFCharsRangeTestCase(768, 879, "Diacritical Marks"),
+                new UTFCharsRangeTestCase(880, 1023, "Greek and Coptic"),
+                new UTFCharsRangeTestCase(1024, 1279, "Cyrillic Basic"),
+                new UTFCharsRangeTestCase(1280, 1327, "Cyrillic Supplement"),
+                new UTFCharsRangeTestCase(8192, 8303, "General Punctuation"),
+                new UTFCharsRangeTestCase(8352, 8399, "Currency Symbols"),
+                new UTFCharsRangeTestCase(8448, 8527, "Letterlike Symbols"),
+                new UTFCharsRangeTestCase(8592, 8703, "Arrows"),
+                new UTFCharsRangeTestCase(8704, 8959, "Mathematical Operators"),
+                new UTFCharsRangeTestCase(9472, 9599, "Box Drawings"),
+                new UTFCharsRangeTestCase(9600, 9631, "Block Elements"),
+                new UTFCharsRangeTestCase(9632, 9727, "Geometric Shapes"),
+                new UTFCharsRangeTestCase(9728, 9983, "Miscellaneous Symbols"),
+                new UTFCharsRangeTestCase(9984, 10175, "Dingbats")
+        };
+    }
+
+    
+
+    @Test
+    @UseDataProvider("getTestCases")
+    public void Test_BytesToChars_No_Middle_String_Conversion(final UTFCharsRangeTestCase testCase) throws IOException {
+        final AppsAPIImpl impl = new AppsAPIImpl();
+        final StringBuilder stringBuilder = new StringBuilder();
+        for(int i=  testCase.fromCode; i<= testCase.toCode; i++) {
+            final String string = fromCharCode(i);
+            stringBuilder.append(string);
+        }
+        final String input = stringBuilder.toString();
+        Logger.info(AppsAPIImplTest.class,()->  String.format(" UTF Charset code from `%d` to `%d`  %s `%s` ",testCase.fromCode, testCase.toCode, testCase.description,input));
+        final char [] chars = impl.bytesToCharArrayUTF(input.getBytes(StandardCharsets.UTF_8));
+        final byte [] bytes = impl.charsToBytesUTF(chars);
+        final String output = new String(bytes, StandardCharsets.UTF_8);
+        assertEquals(input, output);
+    }
+
+    /**
+     * https://www.w3schools.com/charsets/ref_html_utf8.asp
+     * @param codePoints char code see utf char codes.
+     * @return the utf string representation.
+     */
+    private static String fromCharCode(final int... codePoints) {
+        return new String(codePoints, 0, codePoints.length);
+    }
+
+    @Test
+    public void Test_Secret_Json_Serialization_No_String_Middle_Man() throws DotDataException {
```

Comment:
doc?

Reply:
The name is pretty self-explanatory.

Reply:
we usually java doc tests like this
```
/* 
Given scenario:
Expected result: 
*/
```

In this case I don't understand what is the expected result from the name

Reply:
done